graph_explorer/src/graph_explorer/graph_visualizer/views.py

```diff
@@ -145,11 +146,12 @@ def add_filter(request):
     try:
         attribute: str = request.POST["attribute"]
         operator: str = request.POST["operator"]
-        value: str = request.POST["value"]
+        unfiltered_graph = content_module.get_current_workspace().get_unfiltered_graph()
+        value: Any = coerce_filter_value(request.POST["value"], attribute, unfiltered_graph, operator)
         current_workspace = content_module.get_current_workspace()
         operator_filter: OperatorFilter = OperatorFilter(attribute, operator, value)
         current_workspace.add_filter(operator_filter)
-    except (KeyError, ValueError):
-        return
+    except (KeyError, ValueError, TypeError) as e:
+        return HttpResponse(str(e), content_type="text/plain", status=400)
 
-    return HttpResponseRedirect(reverse("index"))
+    return HttpResponseRedirect(reverse("workspace", kwargs={"workspace_id": current_workspace.id}))
```

Comment:
You may leave this as is because in #61 the index will redirect to workspace/active-id